### ISSUE TITLE
UI lags on big repository

### ISSUE THREAD

**Issue by akaRem** [nikhilkalige/git-log#67]
When repository is big, then scrolling becomes laggy and annoying

**Comment by hendricha** [comment]
+1

**Comment by maxpolski** [comment]
I can say, that it almost doesn't work on a huge repo

**Comment by Lknechtli** [comment]
Takes about 15 seconds to load on a repo with 4700 commits, and about a second delay when doing any action such as scrolling.  Would make a lot of sense if it loaded the latest X commits, and loaded more as necessary.

**Comment by mpe** [comment]
The UI never even loads for me. My repo has 574,184 commits, you can clone it from kernel.org.

**Comment by maxpolski** [comment]
Seems, that developers doesn't care about it

**Comment by brunolemos** [comment]
Yeah, impossible to use unfortunately. Thoughts about it @nikhilkalige?

**Comment by programmador** [comment]
+1

**Comment by harmoney-jontie** [comment]
+1
